### PR TITLE
29 game frontend

### PR DESCRIPTION
**Pull request by basheer421**

### CODE REVIEW

**Comment by Salem-1** [comment]
Was not able to play the game, got many exceptions upon running the game in tournament and AI, here are the exceptions logs for our reference :
game.js:1 Uncaught SyntaxError: Identifier 'BALL_SPEED' has already been declared (at game.js:1:1)
tournament_algorithm.js:49 Uncaught (in promise) ReferenceError: playGame is not defined
    at playFinals (tournament_algorithm.js:49:18)
    at startTournament (tournament_algorithm.js:82:24)
    at HTMLButtonElement.onclick (tournament:1:1)
playFinals @ tournament_algorithm.js:49
startTournament @ tournament_algorithm.js:82
onclick @ tournament:1
await in onclick (async)
onclick @ tournament:1
Aigame.js:1 Uncaught SyntaxError: Identifier 'PADDLE_SPEED' has already been declared (at Aigame.js:1:1)
Navigated to http://localhost:3000/home
6Fetch finished loading: GET "<URL>".
Aigame.js:19 Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'.
    at new Paddle (Aigame.js:19:30)
    at Aigame.js:145:12

**Comment by basheer421** [comment]
@Salem-1 Done, please review.

**Comment by Salem-1** [comment]
Tournament notes:
-2, 3, 4 players: no tournament chart shown, they play game again and again, and when press back it doesn't go back
-4 players: they play right away but after winning it doesn't show winner player, it shows garbage names from last rounds
-6 players: it show tournament chart with correct order, but when refresh the tourament start right away without clicking start tournament and the back and front buttons in the browser are disabled
-7 Players: start right away without chart, and keep playing games, there no way to go back except by changing the url manually
-8 players: it show the tournament chart in the qurater finals part, however all other placeholders hold garbage values from old tournaments
-8:players "launch tournament" button doesn't work, with error thrown:
   tournament_algorithm.js:32 Uncaught (in promise) Error: error: fetching players for the tournament
    at startTournament (tournament_algorithm.js:32:15)
    at HTMLButtonElement.onclick (tournament:1:1)

-localhost:300/game: at beging it work, when someone wins it doesn't end.
here comes the fun part:
when going localhost:3000/game then on browser click back then front then back and front, we go back to home page however the game will be funning in the background and every 2 seconds will show alert "player 1 wins" or "player 2 wins", and through the following errors in the console:
VM1416 game.js:102 Uncaught TypeError: Cannot set properties of null (setting 'textContent')
    at updateScore (VM1416 game.js:102:28)
    at draw (VM1416 game.js:155:13)
    at VM1416 game.js:294:17
updateScore @ VM1416 game.js:102
draw @ VM1416 game.js:155
(anonymous) @ VM1416 game.js:294
setInterval (async)
(anonymous) @ VM1416 game.js:293
playGame @ VM1416 game.js:292
main @ VM1416 game.js:316
game @ VM1416 game.js:348
(anonymous) @ VM1416 game.js:351
VM1386 game.js:102 Uncaught TypeError: Cannot set properties of null (setting 'textContent')
    at updateScore (VM1386 game.js:102:28)
    at draw (VM1386 game.js:159:13)
    at VM1386 game.js:294:17
updateScore @ VM1386 game.js:102
draw @ VM1386 game.js:159
(anonymous) @ VM1386 game.js:294
setInterval (async)
(anonymous) @ VM1386 game.js:293
playGame @ VM1386 game.js:292
main @ VM1386 game.js:316
game @ VM1386 game.js:348
(anonymous) @ VM1386 game.js:351
VM1389 game.js:102 Uncaught TypeError: Cannot set properties of null (setting 'textContent')
    at updateScore (VM1389 game.js:102:28)
    at draw (VM1389 game.js:159:13)
    at VM1389 game.js:294:17
updateScore @ VM1389 game.js:102
draw @ VM1389 game.js:159
(anonymous) @ VM1389 game.js:294
setInterval (async)
(anonymous) @ VM1389 game.js:293
playGame @ VM1389 game.js:292
main @ VM1389 game.js:316
game @ VM1389 game.js:348
(anonymous) @ VM1389 game.js:351
VM1392 game.js:102 Uncaught TypeError: Cannot set properties of null (setting 'textContent')
    at updateScore (VM1392 game.js:102:28)
    at draw (VM1392 game.js:159:13)
    at VM1392 game.js:294:17
updateScore @ VM1392 game.js:102
draw @ VM1392 game.js:159
(anonymous) @ VM1392 game.js:294
setInterval (async)
(anonymous) @ VM1392 game.js:293
playGame @ VM1392 game.js:292
main @ VM1392 game.js:316
game @ VM1392 game.js:348
(anonymous) @ VM1392 game.js:351
VM1395 game.js:102 Uncaught TypeError: Cannot set properties of null (setting 'textContent')
    at updateScore (VM1395 game.js:102:28)
    at draw (VM1395 game.js:159:13)
    at VM1395 game.js:294:17
    
-The game: when the ball hit little under the right paddle it doesn't count as goal, however it bounce back normally to the other player

-the winner is whon before the last point scored

-line 100, 101, 176, 228, 2:  getElemntbyIdn() is not protected, the return value might be null, wcich will cause uncought exception 
-in Tournament_algorithm.js all get elemetbyid is not protected as wll

File: .gitignore
Comment (MahdyZ7) [inline .gitignore:1]: valut data should not be ignored

**Review by MahdyZ7 — COMMENTED** [review]
- ball starts form the same postion every time
- 5 player tournament logic should be 3 semifinalists and 2 quarter finalists
- Ai loses his logic if game window change mid game
- ball gets stuck to paddel
- ball registers a hit bellow the paddel by some margin
- ball change spee abrubtly when hitting the top edge but not the bottom

**Pull request merged**